lib/debugger/clients/explorer.js

```diff
@@ -36,38 +34,100 @@ WdRepl.prototype.initClient_ = function() {
 };
 
 /**
- * Eval function for processing a single step in repl.
+ * Instantiate a server to handle IO.
+ * @param {number} port The port to start the server.
  * @private
- * @param {string} cmd
- * @param {object} context
- * @param {string} filename
- * @param {function} callback
  */
-WdRepl.prototype.stepEval_ = function(cmd, context, filename, callback) {
-  cmd = cmd.slice(1, cmd.length - 2);
-  this.cmdRepl.stepEval(cmd, callback);
+WdRepl.prototype.initServer_ = function(port) {
+  var net = require('net');
+  var self = this;
+  var cmdRepl = new CommandRepl(this.client);
+
+  var received = '';
+  net.createServer(function(sock) {
+    sock.on('data', function(data) {
+      received += data.toString();
+      var i = received.indexOf('\r\n');
+      if (i == 0) {
+        return;
+      }
+      var input = received.substring(0, i);
+      received = received.substring(i + 2);
+      if (data[0] === 0x1D) {
+        // '^]': term command
+        self.client.req({command: 'disconnect'}, function() {
+          // Intentionally blank.
+        });
+        sock.end();
+      } else if (input[input.length - 1] === '\t') {
+        // If the last character is the TAB key, this is an autocomplete
+        // request. We use everything before the TAB as the init data to feed
+        // into autocomplete.
+        input = input.substring(0, input.length - 1);
+        var callback = function(err, res) {
+          if (err) {
+            sock.write('ERROR: ' + err + '\r\n');
+          } else {
+            sock.write(JSON.stringify(res) + '\r\n');
+          }
+        };
+        cmdRepl.complete(input, callback);
+      } else {
+        // Normal input
+        input = input.trim();
+        var callback = function(err, res) {
+          if (err) {
+            sock.write('ERROR: ' + err + '\r\n');
+            return;
+          }
+
+          if (res === undefined) {
+            res = '';
+          }
+          sock.write(res + '\r\n');
+        }
+        cmdRepl.stepEval(input, callback);
+      }
+    });
+  }).listen(port);
+
+  console.log('Server listening on 127.0.0.1:' + port);
```

Comment:
I'm reluctant to change the style of console.log because it's inconsistent with how we've been using it so far. Are there any functional reasons to do this?